### ISSUE TITLE
Method and kwarg names are not descriptive

### ISSUE THREAD

**Issue by cisko3000** [coldtype/coldtype#124]
Hello, I was exploring this tool and running examples and found that the code could be better written. For example, a lot of the methods and keyword arguments are single letters and as a result, not descriptive. There is this cheat sheet because of it: https://coldtype.goodhertz.com/cheatsheets/oneletters.html

I know it seems convenient to opt-in for making single character methods and variables, but code is read far more times than it is written. Are you guys open to accepting pull requests that change this?

**Comment by stenson** [comment]
Hi @cisko3000, thanks for filing the issue — what you're saying definitely makes sense, and I think, generally speaking, is good advice to follow. But in the case of the kind of creative coding that coldtype makes possible, I've found the reverse to be true: the code is written much more often than it is read. Or at least that's very true of how I use coldtype. Most software engineering focuses on "long-term" code that is maintained and tweaked over long periods of time. Creative code is usually single files that are written over the course of a few days, in order to make some creative artifact, like a single image or a video, and then revisited only for the purposes of learning from past work.

So while I definitely agree that Coldtype is difficult to learn and the single-letter abbreviations are not descriptive/intuitive, the upside outweighs those downsides. Being able to write code quickly as part of a creative process has been life-changing for me, and makes me want to write more code / code more animations. I tend to think of it a little like learning a new musical instrument — very difficult at first and requiring some rote memorization, but after a while it becomes second nature and leads to more creativity.

That said, better documentation is definitely something I need to work on!

**Comment by stenson** [comment]
Yeah that's a no, I don't want to change the abbreviation style of code in coldtype, which is pretty fundamental to how everything is written. (Though almost all of the abbreviated things are already aliased to longer non-abbreviated versions.) Just out of curiosity, can I ask why you're interested in coldtype then and not something like drawBot? DrawBot has great straightforward method names and documentation.

**Comment by cisko3000** [comment]
Because it can be animated by midi files.
